stable/redis-ha/templates/redis-server-deployment.yaml

```diff
@@ -0,0 +1,50 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  # Pay attention to the redis-role label at runtime. The self-determination logic in the image sets this value accordingly.
+  name: {{ template "fullname" . }}-server
+  labels:
+    release: "{{ .Release.Name }}"
+    app: "redis-ha"
+    name: "redis-server"
+    redis-node: "true"
+{{ include "labels.standard" . | indent 4 }}
+spec:
+  replicas: {{ .Values.replicas.servers }}
+  template:
+    metadata:
+      labels:
+        app: "redis-ha"
+        name: "redis-server"
+        redis-node: "true"
+{{ include "labels.standard" . | indent 8 }}
+    spec:
+      {{- if .Values.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.nodeSelector | indent 8 }}
+      {{- end }}
+      {{- if .Values.tolerations }}
+      tolerations:
+{{ toYaml .Values.tolerations | indent 8 }}
+      {{- end }}
+      containers:
+      - name: redis
+        image: {{ .Values.redis_image }}
+        resources:
+{{ toYaml .Values.resources.server | indent 10 }}
+        env:
+          - name: REDIS_SENTINEL_SERVICE_HOST
+            value: "redis-sentinel"
+          - name: REDIS_CHART_PREFIX
+            value: {{ template "fullname" . }}-
+        ports:
+          - containerPort: 6379
+        volumeMounts:
+          - mountPath: /redis-master-data 
+            name: data
+      volumes:
+      - name: data
+        emptyDir: {}
+#        persistentVolumeClaim:
```

Comment:
Is it the case that this should _never_ have storage attached? If so remove these lines and the mounted emptydir volume for data. If so we should change this to a stateful set with a PVC template.

Reply:
Cleaned this up. We cannot use statefulsets because of the orphaned root node issue mentioned above.

Reply:
Removed all references to persistence.